cmd/tempo-vulture/main.go

```diff
@@ -141,19 +142,53 @@ func main() {
 			metricTracesErrors.WithLabelValues("requestfailed").Add(float64(metrics.requestFailed))
 			metricTracesErrors.WithLabelValues("notfound").Add(float64(metrics.notFound))
 			metricTracesErrors.WithLabelValues("missingspans").Add(float64(metrics.missingSpans))
+			metricTracesErrors.WithLabelValues("incorrectresult").Add(float64(metrics.incorrectResult))
 		}
 	}()
 
 	http.Handle(prometheusPath, promhttp.Handler())
 	log.Fatal(http.ListenAndServe(prometheusListenAddress, nil))
 }
 
+func intervalsBetween(start, stop time.Time, interval time.Duration, retention time.Duration) []time.Time {
```

Comment:
It doesn't look like there is a need for this function to return all of the possible intervals, since only 1 is chosen at random (and the rest thrown away).   It seems more readable to go ahead and have this method do that encapsulate that functionality, change the name/signature to something like 
```go
choosePastTimestamp(start, stop time.Time, interval, retention time.Duration) (newStart time.Time, ts time.Time)
```   

Call it like 
```go
startTime, seed := intervals := intervalsBetween(startTime, now, interval, tempoRetentionDuration)
````

Reply:
Sure, I like that.  Return both items that we need.  I'll make an update.

Reply:
I've moved the logic into a separate method, but if you'd prefer a single method I can adjust for that as well.  I'll need to update the tests to make sure this is consistent if we want to make it a single method.